packages/web-config-server/src/aggregator/QueryBuilder.js

```diff
@@ -61,9 +61,18 @@ export class QueryBuilder {
     // Make standard replacements if required
     this.query.startDate = this.getQueryParameter('startDate');
     this.query.endDate = this.getQueryParameter('endDate');
-
+    const dataPeriodType = this.getQueryParameter('dataPeriodType');
     // Define the period, based on the query, the preconfigured period, or the default
-    if (this.replacementValues.period) {
+
+    // 'dataPeriodType' is used when you want to force converting the periods to a specific period type. 
+    // Eg, if 'dataPeriodType' = 'MONTH', force converting to MONTHLY periods '201701;201702;201703;201704;...'
+    
+    // This is useful in the scenario when you are fetching indicator values which has to request the aggregate `analytics.json` endpoint
+    // and we have to send the correct period type that the data is submitted. Normally this is taken care of if the visualisation has the correct time selector,
+    // but if it does not, we have to manually convert the periods to the right type so that `analytics.json` can return the correct data without aggregation.
+    if (this.query.startDate && this.query.endDate && dataPeriodType) {
+      this.query.period = convertDateRangeToPeriodQueryString(this.query.startDate, this.query.endDate, dataPeriodType);
```

Comment:
I wonder if this is the right place for that? Other clients of data broker may want to fetch DHIS2 indicators, in which case they have to repeat this logic. An alternative would be pass through the query parameter `dataPeriodType` and have data broker pass that to DhisApi, and have it convert there if it is an indicator. I can't remember whether it's easy to pass config all the way through though...

Reply:
Yep that's a good point about other clients may want to fetch DHIS2 indicators, Igor also raised a similar concern so I'll try to do that in DhisApi instead. My only justification is that I thought DhisApi shouldn't need to have any logic to modify the query parameters, meaning that the clients should be responsible for passing in the right query parameters.

Reply:
Ah yep that's a good point. However, I think what we're doing here is making DhisApi behave as expected (i.e. in line with the other endpoints) given certain query parameters, so it doesn't matter so much that it's altering those query parameters as they go through to DHIS2. I do see that it might be surprising for people if they expect DhisApi to act as per the DHIS2 docs...but this use case seems worth it.